ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionClientHandlerImpl.java

```diff
@@ -66,7 +70,10 @@ public SafeFuture<ForkChoiceUpdatedResult> engineForkChoiceUpdated(
             .build();
 
     return methodsResolver
-        .getMethod(EngineApiMethods.ENGINE_FORK_CHOICE_UPDATED, ForkChoiceUpdatedResult.class)
+        .getMilestoneMethod(
+            EngineApiMethod.ENGINE_FORK_CHOICE_UPDATED,
+            () -> spec.atSlot(forkChoiceState.getHeadBlockSlot()).getMilestone(),
```

Comment:
I think should use `PayloadBuildingAttributes.blockSlot` if present, otherwise this slot. Because if we have the first slot of the new milestone missed, this will break if we are about to produce a block for the upcoming milestone (the V2 is actually due to the V2 on the payload attributes param)

Reply:
That makes sense. Will change tomorrow.